README.md

```diff
@@ -1,18 +1,84 @@
 
-# What Does TubiPlayer Do Well.
+## What Does TubiPlayer Do Well.
 * Native multi-media streaming on Android
 * Industry standard Ad solutions, including VAST and VPAID
 * Error-handing
 * Seamless UX 
 * Highly customized module
 
 
-# State Machine Diagram
+## Dependency Management
+In order to provide highly customized code base, we have leveraged [Dagger's](https://github.com/google/dagger) dependency management framework to better manage dependencies for your own business logic.(If you are not familiar with using dagger, please take sometime to learn it, because it is a very good tool to reduce complexity, testing, and other software development merits)
+In general, your business requirement may require different logic of showing video ads at different positions, and different types of ads, and so forth. The idea behind Tubiplayer's dependency management is that we provide the basic classes and functionality
+to implement playback experience, and if you have different logic, you just override that particular class or functionality without the need to thoroughly understand the whole code base. As the result, you are able to easily and quickly use TubiPlayer without invest much time 
+upfront.
+
+Nearly every business related dependencies are managed by [PlayerModuleDefault](./lib/src/main/java/com/tubitv/media/di/PlayerModuleDefault.java), where all the dependencies are being instantiated and being injected into [DoubleViewTubiPlayerActivity](./lib/src/main/java/com/tubitv/media/activities/DoubleViewTubiPlayerActivity.java)
+You can choose to use the default behavior by using the module out of box, but if you have customized business logic, you can directly change the dependencies in ***PlayerModuleDefault*** for customization
+
+## How to use
+Different application often requires different sets of rules and logic to satisfy business requirements, as the result, Tubiplayer was build to provide high degree of
+customization. However, highly customized code if not managed well can quickly turn into nightmare. Therefore, Tubiplayer leverages a third party Dependency Injection framework:[Dagger](https://github.com/google/dagger) 
+to provide the best of both worlds.(If you are not familiar with using dagger, please take sometime to learn it, because it is a very good tool to reduce complexity, testing, and other software development merits)
+
+### Different use cases:
+1. If you just simply want to play a video without any interruption, you can just use ***DoubleViewTubiPlayerActivity*** out of box, then the code start a self contained activity to handle your playback experience
```

Comment:
If you want to play a video without any forms of interruption, you can use ***DoubleViewTubiPlayerActivity*** out of box.  The code start a self contained activity to handle your playback experience.